src/components/text/index.tsx

```diff
@@ -0,0 +1,25 @@
+import React, { FC } from 'react';
+import { Text as ChakraText, TextProps } from '@chakra-ui/react';
+
+type MarginProps = 'm' | 'mt' | 'mb' | 'mr' | 'ml' | 'my' | 'mx';
+type PaddingProps = 'p' | 'pt' | 'pb' | 'pr' | 'pl' | 'py' | 'px';
+
+type Props = Pick<
```

Comment:
Do we really need to expose all these combinations for a text? This allows for very strange combinations I doubt we will ever use in any layout. For example something like this

![image](https://user-images.githubusercontent.com/3371760/183380397-9f27acd6-2ea9-4764-a693-0ed8413a5f27.png)

Reply:
I went for a sensible minimum that gives the flexibility to implement different styles of texts without the need to reach for custom components. What would you omit here?

As the background - I was prompted to add `Text` by [privacy policy and ToC disclaimer in the lead form](https://www.figma.com/file/f0LWQupJueW6KHBQujVGe4/Lead-Form-Page-(Working-Title)?node-id=830%3A71834)

Reply:
I would check back with the designer about what they intend to use/have them create a text component. Alternatively, I would expose what we need and gradually extend it when we need it for other designs while letting the designer know that they are creating a new text variation in their design. My fear is that when being open we will create a lot of small differences between the layouts (line heights, passing etc) so I d rather only provide whats intentionally scoped as a text component

Reply:
I see where this comes from, but I think there's merit to keeping it as is. This is quite a minimal set of props related to text formatting. What I can think of is only exposing `textStyle` and accepting values defined in `typography`. Do you think that's better?

I want to avoid explicitly creating a component/variant for every text style since that would add a significant overhead on our end...

Reply:
as discussed - we drop the `margin` and `padding` (since it can be controlled by the layout), only use `textStyle`, but keep the `color` as is